scripts/protect-branches.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python3
+
+
+import argparse
+import json
+import os
+import re
+import requests
+import sys
+import urllib
+
+
+# API token received from github. For the "scopes", I only selected the entire "repo" section.
+# https://github.com/settings/tokens
+TOKEN=""
```

Comment:
I think using getpass be good and I could help it the stdout the URL to get
the token.  that would be quick I think

On Aug 12, 2016 5:27 PM, "Michael Hrivnak" <notifications@github.com> wrote:

> In scripts/protect-branches.py
> <https://github.com/pulp/devel/pull/3#discussion_r74663004>:
>
> > @@ -0,0 +1,97 @@
> > +#!/usr/bin/env python3
> > +
> > +
> > +import argparse
> > +import json
> > +import os
> > +import re
> > +import requests
> > +import sys
> > +import urllib
> > +
> > +
> > +# API token received from github. For the "scopes", I only selected the entire "repo" section.
> > +# https://github.com/settings/tokens
> > +TOKEN=""
>
> I thought about this too. It's a little inconvenient to set an environment
> variable without putting the value in your shell history, but it's doable.
> I could use getpass() perhaps which would be secure, but would make you
> enter it every time. Or both.
>
> Part of my dilemma is that I wrote this little script to save time and
> thus tried to avoid feature creep, but I agree this is worth considering.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/pulp/devel/pull/3/files/8163f5e3ba09a1c1695d395ef9a5c013076f0fbc#r74663004>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AALAXyUG6s5KODrQ-sHQirK-yDGBiFuyks5qfOUmgaJpZM4Jjaa5>
> .
>